### PR TITLE
added unit test and caching of pixels for speed

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

File: src/common/TestImageUdder.h [hunk 1]
```diff
@@ -20,10 +20,34 @@ class Udder {
...
+      }
+      if (pixel != 0) {
+        CachedPixels.push_back(pixel);
+      }
+      old = pixel;
+    }
+  }
+
 
   uint32_t getPixel(uint16_t w, uint16_t h, ESSGeometry * geom) {
```
Comment (amues) [inline src/common/TestImageUdder.h]: @mortenjc, what is the bit depth for the image?

File: src/common/test/TestImageUdderTest.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  ASSERT_TRUE(udder.isCached());
+  auto pixel = udder.getPixel(udder.ImageWidth, udder.ImageHeight, &geom);
+  ASSERT_EQ(pixel, 123);
+
+  pixel = udder.getPixel(udder.ImageWidth, udder.ImageHeight, &geom);
+  ASSERT_EQ(pixel, 124);
+}
+
+TEST_F(TestImageUdderTest, CachePixelsWrap) {
+  uint32_t FirstPixel, LastPixel;
```
Comment (amues) [inline src/common/test/TestImageUdderTest.cpp]: Variable name is not consistent, as `udder` and `geom` are lowercase, while `FirstPixel` and `LastPixel` are camel case.

File: src/common/TestImageUdder.h (hunk 1 above)
Comment (mortenjc) [reply src/common/TestImageUdder.h]: Point taken, I'll add more comments for sure. The image has a 1-bit depth.

File: src/common/test/TestImageUdderTest.cpp (hunk 2 above)
Comment (mortenjc) [reply src/common/test/TestImageUdderTest.cpp]: Ok, not really sure I care about this in the unit tests, but since it is you, I'll fix this ;-)

**Comment by mortenjc** [comment]
Hmm, the asserts are not new, but I have seen this before. On my mac there is not 
a problem and this is caught by one of the Jenkins builds. I'll fix this.

**Comment by amues** [comment]
The issue is only caught when trying to compile the new test.

**Pull request merged**